### ISSUE TITLE
NVDA doesn't regain focus if the Firefox Save As dialog is invoked

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#2497]
**Reported by elliott94 on 2012-06-27 12:25**
Steps to repro:

1. Go to the NVDA Snapshots download page (this is just an example, and this issue isn't specific to that particular page).
2. On the most recent snapshot, right click on the main package, and choose the "Save Link As" menu option.
3. When the dialog appears, press escape to close it, and attempt to use arrows to try and read the page.

Expected: Reading of the page should continue as it was doing before the dialog was focused.

Actual: No output is given, and you have to alt-tab away and back into Firefox.

**Comment by jcsteh** [comment]
This is [Mozilla bug 782547](https://bugzilla.mozilla.org/show_bug.cgi?id=782547). There's nothing more we can do here; if/when Mozilla fix this, it will behave as expected.

**Comment by jcsteh** [comment]
This is (finally!) fixed for Firefox 74 (currently available for testing in Firefox Nightly).

**Comment by lukaszgo1** [comment]
@jcsteh Is fix for this supposed to be present in Nightly 74.0a1 (2020-01-14) (64-bit)? It certainly is not fixed for me with this version of Nightly on Windows 7 and NVDA 2019.2.1 when invoking save as dialog and dismissing it with Escape.

**Comment by jcsteh** [comment]
Urg. So it seems this fixed the Print dialog, but not the Save As dialog. I
thought I tested both, but apparently not. I'm not sure why they're
different. I'll look into it.

**Comment by lukaszgo1** [comment]
Just FYI open dialog is also not fixed.

**Comment by jcsteh** [comment]
Open/save dialogs are being addressed in https://bugzilla.mozilla.org/show_bug.cgi?id=1606932. I just submitted a fix which is now awaiting review.

**Comment by jcsteh** [comment]
Open and Save dialogs fixed (really!) for Firefox 75 (currently available for testing in Firefox Nightly).
